Plugin/CheckoutLayoutPlugin.php

```diff
@@ -14,10 +14,36 @@ public function __construct(
         $this->_klaviyoScopeSetting = $klaviyoScopeSetting;
     }
 
+    /**
+     * Checks if logged in user has a default address set, if not returns false.
+     *
+     * @param Magento\Framework\App\ObjectManager $objectManager
+     * @param Magento\Customer\Model\Session $customerSession
+     *
+     * @return Magento\Customer\Model\Address|false
+     */
+    public function _getDefaultAddressIfSetForCustomer(
+        \Magento\Framework\App\ObjectManager $objectManager,
+        \Magento\Customer\Model\Session $customerSession
+    ) {
+        $address = false;
+        if ($customerSession->isLoggedIn()) {
+            $customerData = $customerSession->getCustomer()->getData();
+            $customerFactory = $objectManager->get('\Magento\Customer\Model\CustomerFactory')->create();
```

Comment:
can you inject the CustomerFactory directly instead of injecting and using the ObjectManager to get it?